src/editors/data/redux/requests/reducer.test.js

```diff
@@ -0,0 +1,32 @@
+import { initialState, actions, reducer } from './reducer';
+import { RequestStates, RequestKeys } from '../../constants/requests';
+
+const testingState = {
+  ...initialState,
+  arbitraryField: 'arbitrary',
+};
+describe('requests reducer', () => {
+  test('intial state generated on create', () => {
+    expect(reducer(undefined, {})).toEqual(initialState);
+  });
+  describe('handling actions', () => {
+    const requestsList = ['fetchUnit', 'fetchBlock', 'saveBlock'];
+    const createTestParams = (requestKey) => [
+      [requestKey, 'startRequest', requestKey, { status: RequestStates.pending }],
+      [requestKey, 'completeRequest', { requestKey }, { status: RequestStates.completed, response: undefined }],
+      [requestKey, 'failRequest', { requestKey }, { status: RequestStates.failed, error: undefined }],
+      [requestKey, 'clearRequest', { requestKey }, {}],
+    ];
+    const lifecycletest = (target, action, payload, testValue) => {
+      describe(action, () => {
+        it(`load ${target} from payload`, () => {
+          expect(reducer(testingState, actions[action](payload))).toEqual({
+            ...testingState,
+            [target]: testValue,
+          });
+        });
+      });
+    };
+    requestsList.flatMap(val => createTestParams(val)).map((value) => lifecycletest(...value));
```

Comment:
this is very concise, but a bit hard to read/parse...
Something like this may be easier to read, without un-necessarily returning an output array.
```
requestsList.forEach(requestKey => {
  createTestParams(requestKey).forEach((args) => lifecycleTest(...args));
});
```